tool/site/client/src/pages/sendPreSignPage.tsx

```diff
@@ -88,13 +134,33 @@ async function retrievePrescription(baseUrl: string, prescriptionId: string): Pr
   return getResponseDataIfValid(response, isBundle)
 }
 
-async function sendSignRequest(baseUrl: string) {
+async function sendSignRequest(baseUrl: string, sendPageFormValues: SendPreSignPageFormValues) {
+  await updateEditedPrescriptions(sendPageFormValues, baseUrl)
   const response = await axiosInstance.post<SignResponse>(`${baseUrl}prescribe/sign`)
   const signResponse = getResponseDataIfValid(response, isSignResponse)
   redirect(signResponse.redirectUri)
   return signResponse
 }
 
+async function updateEditedPrescriptions(sendPageFormValues: SendPreSignPageFormValues, baseUrl: string) {
+  if (sendPageFormValues.nominatedOds) {
+    const prescriptions = (await axiosInstance.get(`${baseUrl}prescriptions`)).data as Array<Bundle>
+    prescriptions.forEach(prescription => {
+      const medicationRequests = getMedicationRequestResources(prescription)
+      medicationRequests.forEach(medication => medication.dispenseRequest.performer.identifier.value = sendPageFormValues.nominatedOds)
+    })
+    const newPrescriptions: Array<Bundle> = prescriptions
+      .map(p => Array(parseInt(sendPageFormValues.numberOfCopies)).fill(p).map(p => clone(p))
+      ).flat()
+    newPrescriptions.forEach(p => updateBundleIds(p))
+    await axiosInstance.post(`${baseUrl}prescribe/edit`, newPrescriptions)
+  }
```

Comment:
3 forEachs and 2 maps 🧐 I wonder if we could refactor this to be a little easier to read? A

Also could we move the to level if statement to where is function is called rather than here? Bit weird to call this function which might then fail the check and do nothing

Reply:
refactored and removed the if statement which always evaluated to true, since we populate it with a default value

Reply:
yeah much better thank you